doc/rst/language/spec/statements.rst

```diff
@@ -572,13 +572,17 @@ zipper iteration.
 Zipper Iteration
 ~~~~~~~~~~~~~~~~
 
-When multiple iterators are iterated over in a zipper context, on each
-iteration, each expression is iterated over, the values are returned by
-the iterators in a tuple and assigned to the index, and then statement
-is executed.
-
-The shape of each iterator, the rank and the extents in each dimension,
-must be identical.
+When multiple iterand expressions are traversed in a loop using a
+`zip()` expression, the corresponding values yielded by each iterand
+are combined into a tuple, as represented by the loop's index
+variable(s).  The first iterand in the `zip()` expression is said to
+`lead` the loop's iterations, determining the size and shape of the
+iteration space.  Subsequent expressions `follow` the lead iterand.
+In most cases, these follower iterands are expected to conform to the
+number, and shape, of values yielded by the leader.  For example, if
+the first iterand is a 2D array with `m` rows and `n` columns,
+subsequent iterands will need to support iteration over a 2D `m` x `n`
+space as well.
```

Comment:
In the above, it would be nice to indicate that the multiple iterands are advanced in lock-step. Also, when an iterator returns a ref, the corresponding index variable or tuple component is a ref.

Re "need to support iteration over m*n" : for serial loops, this is a future topic. At least we do not enforce it today: `for i in zip({1..3, 1..2}, {1..6}) do writeln(i);` succeeds. So I suggest either put it in a "note: future work" or move it to Parallel Iterators section https://chapel-lang.org/docs/main/language/spec/iterators.html#parallel-iterators . 

Perhaps a "next steps" item is to expand that Parallel Iterators section with these contents. And/or point that section and this section to each other.

Reply:
> it would be nice to indicate that the multiple iterands are advanced in lock-step

To me, lock-step tends to carry implications of a serial implementation, so I'm reluctant to use that term given that this section is used to define both serial and parallel zippering.  That nature is what I was hoping to express by "corresponding"—i.e., even if element 5 is the first element yielded by a `zip(1..10, A)` loop, the 5th element of A will be yielded with it.

> Also, when an iterator returns a ref, the corresponding index variable or tuple component is a ref.

To me, the ref vs. copy-out nature of iterators and loops seems pretty orthogonal to the concepts in this section—which is to say, if we've defined that appropriately for single-iterand loops, I don't think it needs to be called out again here.  I've changed the word "values" to "expressions", though if that's what was giving you pause.

> this is a future topic. At least we do not enforce it today

I consider this a bug, so don't think we need to document the buggy behavior in the spec.  I've opened #11428 to capture the issue and will file a related future with this PR.